src/app/components/upload-event/upload-event.component.html

```diff
@@ -1,6 +1,6 @@
 <div class="container">
   <h2>Upload an Event</h2>
-  <form (ngSubmit)="uploadEvent()" #eventForm="ngForm">
+  <form (ngSubmit)="uploadEvent();eventForm.reset()" #eventForm="ngForm">
```

Comment:
The form reset should probably happen inside the submit action...

Reply:
Just to clarify I think Marco means add a method to the component  ts file (called something like onSubmit) and do your submit operations in there. You tend to want to keep the logic out of the HTML file where possible.

You'll then either need to pass the event form into the onSubmit method or use ViewChild (https://blog.angular-university.io/angular-viewchild/ )to get a reference.

e.g.

```
<form (ngSubmit)="onSubmit()" #eventForm="ngForm">
```

```
@ViewChild('eventForm') eventForm;

onSubmit(){
  uploadEvent();
  eventForm.reset()
}
```

Reply:
Ahh ok I see! Makes sense to keep it all as "angularly" as possible. Should be helpful for later as well if I ever have to make references to those input elements again.